models.go

```diff
@@ -0,0 +1,32 @@
+package main
+import (
+  "time"
+)
+
+// request and response structures
+
+// defines object returned to status handler
+type StatusResponse struct {
+  Task_id string `json:"task_id"`
+  Finished bool `json:"finished"`
+  StartTime time.Time `json:"start_time"`
+  EndTime   time.Time `json:"end_time,omitempty"`
+  ExitCode  int `json:"exit_code,omitempty"`
```

Comment:
nit: Omitting `ExitCode` will mask the successful execution. This could instead be a pointer to int if you wanted to distinguish between set and unavailable exit code.

Reply:
I didn't notice this while testing, thanks

Reply:
No problem - it is a common issue. Note, that this will not be a problem in this case as unmarshaling the value will simply leave ExitCode uninitialized (and hence at `0`) but like I said there's no way of telling between set `0` and no exit code.